src/script/lua_api/l_mapgen.cpp

```diff
@@ -699,6 +695,24 @@ int ModApiMapgen::l_get_mapgen_object(lua_State *L)
 			lua_setfield(L, -2, it->first.c_str());
 		}
 
+		// push user-defined data
+		auto &custom_map = mg->gennotify.getCustom();
+
+		lua_createtable(L, 0, custom_map.size());
+		lua_getglobal(L, "core");
+		lua_getfield(L, -1, "deserialize");
+		lua_remove(L, -2); // remove 'core'
+		for (auto it : custom_map) {
```

Comment:
```suggestion
		for (const auto &it : custom_map) {
```
I don't think we want to copy `std::pair<std::string, std::string>` instance.